### PR TITLE
Refactor channels

### PR DESCRIPTION
**Pull request by Pugzy**
Continuation of https://github.com/PGMDev/PGM/pull/1306 updated to new cloudy framework

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/channels/ChannelManager.java [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+    return teamChannel;
+  }
+
+  public void setChannel(MatchPlayer player, SettingValue value) {
+    selectedChannel.put(player.getId(), findChannelBySetting(value));
+  }
+
+  public void setChannel(MatchPlayer player, Channel<?> channel) {
+    Channel<?> previous = selectedChannel.get(player.getId());
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/channels/ChannelManager.java]: No need to get & put, put already gives you previous value

File: core/src/main/java/tc/oc/pgm/channels/PrivateMessageChannel.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        sender.getBukkit(), message.substring(1, spaceIndex == -1 ? message.length() : spaceIndex));
+    if (target == null) throw exception("command.playerNotFound");
+
+    if (spaceIndex == -1) {
+      setTarget(selectedPlayer, sender, target);
+      PGM.get().getChannelManager().setChannel(sender, this);
+      return;
+    }
+
+    context.store(MESSAGE_KEY, message.substring(spaceIndex + 1));
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/channels/PrivateMessageChannel.java]: Missing a trim() after the substring

File: core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java [hunk 3]
```diff
@@ -1,488 +1,64 @@
...
+    channel.broadcastMessage(message, null);
+    if (sound.isPresent()) viewers.forEach(viewer -> playSound(viewer, sound.get()));
   }
 
   public static void playSound(MatchPlayer player, Sound sound) {
     SettingValue value = player.getSettings().getValue(SettingKey.SOUNDS);
     if (value.equals(SettingValue.SOUNDS_ALL)
         || value.equals(SettingValue.SOUNDS_CHAT)
-        || (sound.equals(Sounds.DIRECT_MESSAGE) && value.equals(SettingValue.SOUNDS_DM))) {
+        || (sound.equals(DM_SOUND) && value.equals(SettingValue.SOUNDS_DM))) {
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: Does this in chat dispatcher even make sense anymore?

**Review by Pablete1234 — CHANGES_REQUESTED** [review]
I wonder if ChannelManager should just be renamed to ChatManager too

File: core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java [hunk 4]
```diff
@@ -1,488 +1,64 @@
...
 
-  private static final String GLOBAL_SYMBOL = "!";
-  private static final String DM_SYMBOL = "@";
-  private static final String ADMIN_CHAT_SYMBOL = "$";
-
-  private static final String GLOBAL_FORMAT = "<%s>: %s";
-  private static final String PREFIX_FORMAT = "%s: %s";
-  private static final String AC_FORMAT =
-      TextTranslations.translateLegacy(ADMIN_CHAT_PREFIX) + PREFIX_FORMAT;
+  private static final Sound DM_SOUND = sound(key("random.orb"), Sound.Source.MASTER, 1f, 1.2f);
```
Comment (calcastor) [inline core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: Import `Sounds.DIRECT_MESSAGE` from the `Sounds` interface if it's needed; slipped through rebase?

Comment (Pablete1234) [reply core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: chat dispatcher is only kept for backwards-compat, it should simply not have any of this left

File: core/src/main/java/tc/oc/pgm/channels/PrivateMessageChannel.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+public class PrivateMessageChannel implements Channel<MatchPlayer> {
+
+  private static final List<String> ALIASES = List.of("msg", "tell", "r");
+
+  private static final CloudKey<MatchPlayer> TARGET_KEY = CloudKey.of("target", MatchPlayer.class);
+
+  private final OnlinePlayerUUIDMapAdapter<MessageSenderIdentity> selectedPlayer, lastMessagedBy;
+
+  private static final Sound SOUND = sound(key("random.orb"), Sound.Source.MASTER, 1f, 1.2f);
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/channels/PrivateMessageChannel.java]: This however, yeah, it should reference `Sounds` for multi-version support

File: core/src/main/java/tc/oc/pgm/channels/AdminChannel.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+  private static final List<String> ALIASES = List.of("a");
+
+  private static final TextComponent PREFIX = text()
+      .append(text("[", NamedTextColor.WHITE))
+      .append(text("A", NamedTextColor.GOLD))
+      .append(text("] ", NamedTextColor.WHITE))
+      .build();
+
+  private static final Sound SOUND = sound(key("random.orb"), Sound.Source.MASTER, 1f, 0.7f);
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/channels/AdminChannel.java]: Same for this

**Review by Pablete1234 — APPROVED** [review]

**Pull request merged**